### ISSUE TITLE
Fork instruction

### ISSUE THREAD

**Issue by zephyyrr** [zephyyrr/soda#23]
Add a instruction to split execution into another thread.
Memory is shared between the threads and execute independently.

With this, thread-safe memory access operations should be added.
Or the existing operations should be secured, possibly only when multiple threads is running.

**Comment by Victorystick** [comment]
I would prefer threadsafe memory operations/blocks above making all(?)
instructions lock or similar.

**Comment by zephyyrr** [comment]
No, I meant the existing memory operations, not all existing operations.
